src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -651,7 +644,9 @@ static void UpdateCurrentState(EvmState state, int pc, long gas, int stackHead)
             UInt256 localPreviousDest = previousCallOutputDestination;
             if (!UpdateMemoryCost(vmState, ref gasAvailable, in localPreviousDest, (ulong)previousCallOutput.Length))
             {
-                ThrowStackOverflowException();
+                // Never ran any instructions, so nothing to trace (or mark as end of trace)
+                traceOpcodes = false;
+                goto OutOfGas;
```

Comment:
Is StackOverflowException same as OutOfGas?

Reply:
Function was incorrectly named; it threw a `throw new OutOfGasException()`